File: rb/features.rb
# VisitorStatistics SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module VisitorStatisticsFeatures
  def self.make_feature(name)
    case name
    when "base"
      VisitorStatisticsBaseFeature.new
    when "ratelimit"
      VisitorStatisticsRatelimitFeature.new
    when "retry"
      VisitorStatisticsRetryFeature.new
    when "test"
      VisitorStatisticsTestFeature.new
    when "timeout"
      VisitorStatisticsTimeoutFeature.new
    else
      VisitorStatisticsBaseFeature.new
    end
  end
end

File: py/features.py
# VisitorStatistics SDK feature factory

from feature.base_feature import VisitorStatisticsBaseFeature
from feature.test_feature import VisitorStatisticsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VisitorStatisticsBaseFeature(),
        "test": lambda: VisitorStatisticsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
